lib/index.ts

```diff
@@ -81,3 +84,37 @@ function buildDepTreeFromFiles(
       'either packages.config or project file.');
   }
 }
+
+function extractTargetFrameworksFromFiles(
+  root: string, manifestFilePath: string, includeDev = false) {
+  if (!root || !manifestFilePath) {
+    throw new Error('Missing required parameters for extractTargetFrameworksFromFiles()');
+  }
+
+  const manifestFileFullPath = path.resolve(root, manifestFilePath);
+
+  if (!fs.existsSync(manifestFileFullPath)) {
+    throw new Error('No project file found at ' +
+      `location: ${manifestFileFullPath}`);
+  }
+
+  const manifestFileContents = fs.readFileSync(manifestFileFullPath, 'utf-8');
+  const manifestFileExtension = path.extname(manifestFileFullPath);
+
+  if (_.includes(PROJ_FILE_EXTENSION, manifestFileExtension)) {
+    return extractTargetFrameworksFromProjectFile(manifestFileContents);
+  } else {
+    throw new Error(`Unsupported file ${manifestFilePath}, Please provide ` +
```

Comment:
Error text looks misleading as we check `PROJ_FILE_EXTENSION` only

Reply:
thanks!